projects/extension/src/background/index.ts

```diff
@@ -33,22 +35,61 @@ declare let window: Background
 window.manager = publicManager
 
 const l = logger("Extension")
+/**
+ * The amount of minutes that the DatabaseContentAlarm
+ * will rerun
+ */
+const periodInMinutes = 5
+
 export interface RequestRpcSend {
   method: string
   params: unknown[]
 }
 
+const setLocalStorage = async (key: string, chain: Chain) => {
+  const db = await chain.databaseContent(
+    chrome.storage.local.QUOTA_BYTES / wellKnownChains.size,
+  )
+  chrome.storage.local.set({ [key]: db })
+}
+
+const flushDatabases = (): void => {
+  for (const [key, chain] of wellKnownConnections) {
+    setLocalStorage(key, chain)
+  }
+}
+
 const init = async () => {
   try {
     manager = new ConnectionManager(start({ maxLogLevel: 3 }))
     for (const [key, value] of wellKnownChains.entries()) {
       const rpcCallback = (rpc: string) => {
         console.warn(`Got RPC from ${key} dummy chain: ${rpc}`)
       }
-      await manager
-        .addChain(value, rpcCallback)
-        .catch((err) => l.error("Error", err))
+
+      const dbContent = await new Promise<string>((res) =>
+        chrome.storage.local.get([key], (val) => {
+          return res(val[key] as string)
```

Comment:
```suggestion
          res(val[key])
```

Reply:
this change gives an error: `72:22  error  Unsafe argument of type `any` assigned to a parameter of type `string | PromiseLike<string>`  @typescript-eslint/no-unsafe-argument`